core/Common.php

```diff
@@ -263,33 +261,9 @@ public static function mb_strtolower($string)
      */
     public static function sanitizeInputValues($value, $alreadyStripslashed = false)
     {
-        if (is_numeric($value)) {
-            return $value;
-        } elseif (is_string($value)) {
-            $value = self::sanitizeString($value);
-
-            if (!$alreadyStripslashed) {
-                // a JSON array was already stripslashed, don't do it again for each value
-
-                $value = self::undoMagicQuotes($value);
-            }
-        } elseif (is_array($value)) {
-            foreach (array_keys($value) as $key) {
-                $newKey = $key;
-                $newKey = self::sanitizeInputValues($newKey, $alreadyStripslashed);
-                if ($key != $newKey) {
-                    $value[$newKey] = $value[$key];
-                    unset($value[$key]);
-                }
-
-                $value[$newKey] = self::sanitizeInputValues($value[$newKey], $alreadyStripslashed);
-            }
-        } elseif (!is_null($value)
-            && !is_bool($value)
-        ) {
-            throw new Exception("The value to escape has not a supported type. Value = " . var_export($value, true));
-        }
-        return $value;
+        /** @var RequestSanitizer $requestSanitizer */
+        $requestSanitizer = StaticContainer::get('Piwik\Http\RequestSanitizer');
+        return $requestSanitizer->sanitizeValues($value, $alreadyStripslashed);
```

Comment:
Ok, I can't really fix this now. It's due to StaticContainer, but I can't fix it w/o removing StaticContainer. Which I can't do w/o removing static functions like getRequestVar. It's not possible w/o a large commit.

I suggest, before we work on 3.0, we should create some xhprof profiles for specific parts of Piwik that we don't want performance regressions in (like tracking, archiving, all websites/multisites). We post the PHP benchmark code + xhprof profiles online somewhere. Then while working on 3.0, we can re-run the benchmarks periodically and make sure that we're not regressing, or at least how to eventually fix the regression, before the 3.0 release. What do you think?

If you agree, could you list the places we shouldn't regress? You've done most of the performance tuning, I think.